include/revng/Yield/Instruction.h

```diff
@@ -33,7 +48,7 @@ type: struct
 
   - name: Tags
     sequence:
-      type: SortedVector
+      type: detail::TagComparedSortedVector
```

Comment:
No, you're surreptitiously expanding the required expressive power of `TupleTreeGenerator`.
Turn `TagComparator` into `operator<` and let's call it a day.

Reply:
As the result, I can't think of any other way. It's either this or abandoning `SortedVector` here and switching to `std::set` or something. Unless you know some clever trick to override the `Key`'s operator.

Reply:
What's wrong `Tag::Key::operator<`? Why do you need to change it?

Reply:
I need to sort tags in a particular order. And since I'm already using a sorting container to begin with, it sounds like a perfect idea to combine both.

(Except it was a really bad idea: because of it, the tags were broken because, as it turns out, just overloading the operator for the type was not enough and I never noticed)

I guess, there's always an option of keeping them in a normal vector and sorting them manually. But that seems like a lot of extra steps.

Reply:
I suggest to leave `SortedVector` as default and reorder (copying?) when using it.
Push the boundaries of what can be done with `TupleTree` is a losing battle.

Reply:
If I have to resort them manually anyway, why not just use `std::vector` to store them? Or even the earlier `std::set` suggestion for keeping the list sorted?

On that note, why is it that `SortedVector` uses `Type::Key::operator<` as opposed to `Type::operator<`? Was there a specific reason behind it or is it just that no-one complained?

Reply:
`std::vector` and `std::set` are kinda problematic in TTG. Let's not use them unless we really need them.

You're pushing changes into TTG for reason that regard how you intend the data structure in a particular place.
If a random bypasser would not be interested in such ordering, don't put it in the data structure definition. Re-sort them when you need them.

> On that note, why is it that SortedVector uses Type::Key::operator< as opposed to Type::operator<?

I'm inclined to prevent you from customizing any `operator<` in TTG. Let's just do the default thing, so we can replicate in other TTG backends.

tl;dr: unless there's a really really really good reason, TTG is the master, your code is the servant.

Reply:
Sad but gotcha.

Reply:
This is gone :tada: